### PR TITLE
sys.cf_version_release

### PR DESCRIPTION
**Pull request by vpodzime**
Merge together:
https://github.com/cfengine/core/pull/4110
https://github.com/cfengine/buildscripts/pull/707

### CODE REVIEW

File: libenv/sysinfo.c [hunk 1]
```diff
@@ -388,33 +388,29 @@ void DiscoverVersion(EvalContext *ctx)
...
     }
     else
     {
         EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "cf_version_major", "BAD VERSION " VERSION, CF_DATA_TYPE_STRING, "source=agent");
         EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "cf_version_minor", "BAD VERSION " VERSION, CF_DATA_TYPE_STRING, "source=agent");
         EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "cf_version_patch", "BAD VERSION " VERSION, CF_DATA_TYPE_STRING, "source=agent");
-        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "libdir", workdir, CF_DATA_TYPE_STRING, "source=agent");
     }
+
+    EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "cf_release_number", RELEASE, CF_DATA_TYPE_STRING, "source=agent");
```
Comment (olehermanse) [inline libenv/sysinfo.c]: Maybe more appropriate to follow the naming of the others, i.e.

```
sys.cf_version_release
```

or

```
sys.cf_version_patch_release
```

Comment (vpodzime) [reply libenv/sysinfo.c]: Not sure, I'm following the JIRA ticket.

**Comment by vpodzime** [comment]
Custom build:
```
[root@hub vagrant]# cf-promises --show-vars=sys.cf_release_number
Variable name                            Variable value                                               Meta tags                               
default:sys.cf_release_number            5                                                            source=agent                            
```

**Comment by vpodzime** [comment]
Normal build:
```
[root@hub vagrant]# cf-promises --show-vars=sys.cf_release_number
Variable name                            Variable value                                               Meta tags                               
default:sys.cf_release_number            1                                                            source=agent                            
```

File: libenv/sysinfo.c (hunk 1 above)
Comment (vpodzime) [reply libenv/sysinfo.c]: What do you think, @nickanderson?

File: examples/variablesmatching.cf [hunk 2]
```diff
@@ -44,5 +44,6 @@ bundle agent run
 #@ R: Variables matching 'default:sys.cf_version.*' = default:sys.cf_version_major
 #@ R: Variables matching 'default:sys.cf_version.*' = default:sys.cf_version_minor
 #@ R: Variables matching 'default:sys.cf_version.*' = default:sys.cf_version_patch
+#@ R: Variables matching "default:sys.cf_version.*" = default:sys.cf_version_release
```
Comment (olehermanse) [inline examples/variablesmatching.cf]: @vpodzime need to use single quotes here.

Comment (vpodzime) [reply examples/variablesmatching.cf]: Yeah, saw that :disappointed:

Comment (vpodzime) [reply examples/variablesmatching.cf]: The crazy thing is the output from the tests, though:
```
Q: "...r/bin/perl /hom": -R: Variables matching "default:sys.cf_version.*" = default:sys.cf_version_release
Q: "...r/bin/perl /hom": +R: Variables matching "default:sys.cf_version.*" = default:sys.cf_version_release
```
That's where I took the double-quotes from.

Comment (olehermanse) [reply examples/variablesmatching.cf]: @vpodzime it seems cf-agent replaces the quotes from the diff output?

**Pull request merged**